File: backend/src/main/java/com/offerflow/service/ProfileService.java
package com.offerflow.service;

import com.offerflow.dto.request.ChangePasswordRequest;
import com.offerflow.dto.response.UserProfileResponse;
import com.offerflow.entity.User;

public interface ProfileService {

    UserProfileResponse getProfile(
            User user);

    void changePassword(
            ChangePasswordRequest request,
            User user);
}

File: frontend/src/services/profileService.ts
import api from "@/api/axios";

import type {
  Profile,
  ChangePasswordRequest,
} from "@/types/profile";

export async function getProfile() {
  const response =
    await api.get<Profile>(
      "/profile",
    );

  return response.data;
}

export async function changePassword(
  request: ChangePasswordRequest,
) {
  await api.put(
    "/profile/password",
    request,
  );
}